src/preprocessing/passes/bv_to_int.cpp

```diff
@@ -315,9 +315,8 @@ Node BVToInt::bvToInt(Node n)
             }
             else
             {
-              // Boolean variables are left unchanged.
               AlwaysAssert(current.getType() == d_nm->booleanType()
-                           || current.getType().isSort());
+                           || current.getType().isSort() || current.getType() == d_nm->integerType() || current.getType() == d_nm->realType() );
```

Comment:
Can you generalize this?  Shouldn't it be that all types besides bit-vector are preserved?

Reply:
All types in the supported logics besides bit-vectors, except UFs, that need to change also. So I can change it to `current type not in [bitvector, UF]`,  but that would rely on the fact that we are in a supported logic. A bit less safe, and not a lot clearer in my opinion, but if u prefer it that way i can change.

Reply:
As I mentioned, we should generalize to handle other logics.

Reply:
as i responded, that could be problematic :)
https://github.com/CVC4/CVC4/pull/3905#issuecomment-594108026